Sources/BSON/BSONObjectID.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import NIO
+import NIOConcurrencyHelpers
+
+/// A struct to represent the BSON ObjectID type.
+public struct BSONObjectID: Equatable, Hashable, CustomStringConvertible {
+    /// This `BSONObjectID`'s data represented as a `String`.
+    public var hex: String { self.oid.reduce("") { $0 + String(format: "%02x", $1) } }
```

Comment:
some tests verifying that converting an `ObjectID` to/from a hex string happens losslessly would be nice.